services/configuration/src/transactions.rs

```diff
@@ -288,22 +288,23 @@ impl<'a> VotingDecisionRef<'a> {
         }
 
         let schema = Schema::new(snapshot);
+        let propose = schema
+            .propose(self.cfg_hash())
+            .ok_or_else(|| UnknownConfigRef(*self.cfg_hash()))?;
+
         if let Some(validator_id) = validator_index(snapshot, self.sender()) {
             let vote = schema
                 .votes_by_config_hash(self.cfg_hash())
                 .get(validator_id as u64)
-                .unwrap();
+                .expect("Can't get vote for precheck");
+
             if vote.is_some() {
-                Err(AlreadyVoted)?;
+                return Err(AlreadyVoted);
```

Comment:
Nit: Is there a particular reason you've changed this instance of erroneous return, but didn't change the one above, with `Err(AlreadyScheduled(_))`?

Reply:
There are too much of similar entries in this file, so I guess I'll create a separate pull-request for this.